include/boost/url/impl/decoded_view.ipp

```diff
@@ -0,0 +1,140 @@
+//
+// Copyright (c) 2022 Alan de Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/CPPAlliance/url
+//
+
+#ifndef BOOST_URL_IMPL_DECODED_VIEW_IPP
+#define BOOST_URL_IMPL_DECODED_VIEW_IPP
+
+#include <boost/url/decoded_view.hpp>
+#include <ostream>
+
+namespace boost {
+namespace urls {
+
+decoded_view::
+decoded_view(
+    string_view str,
+    pct_decode_opts opt) noexcept
+    : p_(str.data())
+    , n_(str.size())
+    , plus_to_space_(opt.plus_to_space)
+{
+    error_code ec;
+    opt.non_normal_is_error = false;
+    dn_ = validate_pct_encoding(
+        str, ec, opt,
+        [](unsigned char c) { return c != '%'; });
+    if (ec.failed())
+        detail::throw_invalid_argument(
+            BOOST_CURRENT_LOCATION);
+}
+
+decoded_view::size_type
+decoded_view::
+copy(
+    char* dest,
+    size_type count,
+    size_type pos) const
+{
+    if( pos > size() )
+        detail::throw_invalid_argument(
+            BOOST_CURRENT_LOCATION);
+    std::size_t rlen = (std::min)(count, size() - pos);
+    auto first = begin();
+    while (pos--)
+        ++first;
```

Comment:
`first = std::next( first, pos );`

Reply:
Yes.

It would probably be 

```cpp
auto first = std::next( begin(), pos );
```

or 

```cpp
auto first = begin();
std::advance( first, pos );
```

I thought you didn't want me to use these functions to avoid the includes. But the include is there anyway because of the iterator categories.